gelbooru-overhaul.user.js

```diff
@@ -9,485 +9,1551 @@
+                            general: [...htmlDocument.querySelectorAll(".tag-type-general     > a")].map(i => i.text),
+                        };
+
+                        if (!highResThumb || !fileLink) throw new Error("Failed to parse url");
+
+                        postCache[postId] = {
+                            highResThumb: highResThumb,
+                            download: fileLink,
+                            tags: tags,
+                            md5: md5,
+                            id: postId
+                        };
+                        GM_setValue("postCache", postCache);
+                        resolve(postCache[postId]);
+                    })
+                    .catch(error => reject(error));
+            } else {
+                resolve(postCache[postId]);
             }
-            //system in dark mode
-            if(window.matchMedia('(prefers-color-scheme: dark)').matches)
-            {
-                return true;
+        });
+    }
+    /**
+     * 
+     * @param {PostItem} post 
+     * @returns {Promise}
+     */
+    function downloadPostItem(post) {
+        return new Promise((resolve, reject) => {
+            //build filename
+            let filename = String(configManager.config.fastDL.items.pattern.value);
+            let spr = String(configManager.config.fastDL.items.separator.value);
+
+            filename = filename.replace("%md5%", post.md5);
+            filename = filename.replace("%postId%", String(post.id));
+            filename = filename.replace("%artist%", post.tags.artist.length ? post.tags.artist.join(spr) : "unknown_artist");
+            filename = filename.replace("%character%", post.tags.character.length ? post.tags.character.join(spr) : "unknown_character");
+            filename = filename.replace("%copyright%", post.tags.copyright.length ? post.tags.copyright.join(spr) : "unknown_copyright");
+
+            filename = filename.replace(/[<>:"/\|?*]/, "_"); // illegal chars
+
+            GM_download({
+                url: post.download,
+                name: filename + "." + post.download.split(".").at(-1),
+                saveAs: configManager.config.fastDL.items.saveAs.value,
+                onload: resolve("Download finished"),
+                onerror: (error, details) => reject({ error, details }),
+            });
+
+            if (configManager.config.fastDL.items.saveTags.value) {
+                let text = post.tags.artist.map(i => "artist:" + i).join("\n") + "\n" +
+                    post.tags.character.map(i => "character:" + i).join("\n") + "\n" +
+                    post.tags.copyright.map(i => "series:" + i).join("\n") + "\n" +
+                    post.tags.metadata.map(i => "meta:" + i).join("\n") + "\n" +
+                    post.tags.general.join("\n") + "\n";
+
+                text = text.replaceAll("\n\n", "\n");
+
+                let elem = document.createElement("a");
+                elem.href = "data:text," + text;
```

Comment:
```suggestion
                elem.href = "data:text," + encodeURIComponent(text);
```
Fix newlines in the FastDL text file not working in Firefox.

*(Confirmed working on Tampermonkey 4.17.6161 for Chrome and Firefox, but maybe quadruple-check that this screw up any edge-cases)*